plugins/dialog/plugin.js

```diff
@@ -500,6 +503,150 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
+				( !isCurrentNodeInRadioGroup && isPreviousNodeInRadioGroup )
+			) {
+				return currentIndex;
+			}
+
+			// If the current index points to the radio group, find the checked element and
+			// move focus to it.
+			if ( !isPreviousNodeInRadioGroup && isCurrentNodeInRadioGroup ) {
+				var radioGroup = getRadioGroup( currentIndex, direction ),
+					focusedItemIndex = getFocusedRadioGroupElement( radioGroup );
+
+				// If no radio button is checked then:
+				// When Tab was pressed, then focus the first radio button.
+				// When Shift+Tab was pressed, then focus the last radio button.
+				if ( focusedItemIndex === -1 ) {
+					focusedItemIndex = 0;
+					dialogUiElements[ currentIndex ].getInputElement().$.focus();
+				}
+
+				currentIndex = updateCurrentIndexBy( currentIndex, focusedItemIndex, direction );
+
+				return currentIndex;
+			}
+
+			// If the previous and the current element is in the radio group then find the
+			// new radio group and move focus to selected element.
+			if ( isPreviousNodeInRadioGroup && isCurrentNodeInRadioGroup ) {
+				var groupId = dialogUiElements[ startIndex ]
+					.getElement()
+					.getAscendant( 'table' )
+					.getId();
+
+				while ( isWithinTheSameRadioGroup( groupId, currentIndex ) ) {
+					currentIndex = updateCurrentIndex( currentIndex, direction );
+				}
+
+				var radioGroup = getRadioGroup( currentIndex, direction );
+
+				if ( radioGroup.length > 0 ) {
+					var focusedItemIndex = getFocusedRadioGroupElement( radioGroup );
+
+					if ( focusedItemIndex === -1 ) {
+						focusedItemIndex = 0;
+						dialogUiElements[ currentIndex ].getInputElement().$.focus();
+					}
+
+					currentIndex = updateCurrentIndexBy( currentIndex, focusedItemIndex, direction );
+				}
+			} else {
+				var radioGroupLength = dialogUiElements[ startIndex ].items.length;
+				// When focus is on radio button, Tab or Shift+Tab will exit the radio group or
+				// move to the next radio group.
+				for ( var i = 0; i < radioGroupLength - 1; i++ ) {
+					currentIndex = updateCurrentIndex( currentIndex, direction );
+				}
+			}
+
+			// When the current element is not focusable then depends on direction
+			// get the next or previous element.
+			if ( !dialogUiElements[ currentIndex ].isFocusable() ) {
+				while ( !dialogUiElements[ currentIndex ].isFocusable() ) {
+					currentIndex = updateCurrentIndex( currentIndex, direction );
+				}
+			}
+
+			return currentIndex;
+		}
+
+		function getRadioGroup( startIndex, direction ) {
+			var dialogUiElements = me._.focusList,
+				radioGroup = [],
+				length;
+
+			if ( dialogUiElements[ startIndex ].items ) {
+				length = dialogUiElements[ startIndex ].items.length;
+			}
+
+			if ( length === undefined ) {
+				return radioGroup;
+			}
+
+			for ( var i = 0; i < length; i++ ) {
+				radioGroup.push( dialogUiElements[ startIndex ] );
+				startIndex = updateCurrentIndex( startIndex, direction );
```

Comment:
The getter shouldn't update any status, but from what I see, it actually does not update anything (just the name `updateCurrentIndex` is misleading).